api/envoy/api/v2/core/grpc_method_list.proto

```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
+
+package envoy.api.v2.core;
+
+import "udpa/annotations/migrate.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.api.v2.core";
+option java_outer_classname = "GrpcMethodListProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_migrate).move_to_package = "envoy.config.core.v3";
+
+// A list of gRPC methods which can be used as a whitelist, for example.
+message GrpcMethodList {
+  message Service {
+    // The name of the gRPC service.
+    string name = 1 [(validate.rules).string = {min_bytes: 1}];
+
+    // The names of the gRPC methods in this service.
+    repeated string method_names = 2 [(validate.rules).repeated = {min_items: 1}];
```

Comment:
@htuch @mattklein123 @kyessenov Do any of you have a preference on this structure, vs just having a repeated message with two strings (service, method), with the service name being duplicated for each method within it?

Reply:
I don't have a preference. Either way is fine with me.

Reply:
No preference, but grouping by service seems more efficient.